src/main/scala/se/nimsa/dcm4che/streams/DicomParsing.scala

```diff
@@ -55,6 +72,82 @@ trait DicomParsing {
     }
   }
 
+  // parse dicom UID data element from buffer
+  def parseUIDDataElement(data: ByteString, explicitVR: Boolean, assumeBigEndian: Boolean): DataElement = {
+    val maybeHeader = if (explicitVR) {
+      readHeaderExplicitVR(data, assumeBigEndian)
+    } else {
+      readHeaderImplicitVR(data)
+    }
+
+    if (maybeHeader.isEmpty) {
+      throw new DicomStreamException("Could not parse DICOM data element from stream.")
+    }
+
+    val (tag, vr, headerLength, length) = maybeHeader.get
+    val value = data.drop(headerLength).take(length)
+    DataElement(tag, vr, length, valueWithoutPadding(value))
+  }
+
+
+  def valueWithoutPadding(value: ByteString) =
```

Comment:
This method may only be used if you know the length of the input value, right? In other words, in cases where you know the value is padded? If not, you may be stripping a trailing zero that had some meaning. Just making sure I follow.